pynestml/codegeneration/resources_nest/point_neuron/directives/ReadFromDictionaryToTmp.jinja2

```diff
@@ -19,9 +22,24 @@ if (tmp_{{names.name(variable)}} != {{names.getter(variable)}}())
 {%- endfor %}
 }
 {%- endif %}
-{%- elif not variable.is_inline_expression and variable.is_state() %}
+
+{%- else %}
 {{declarations.print_variable_type(variable)}} tmp_{{names.convert_to_cpp_name(variable.get_symbol_name())}} = {{names.getter(variable)}}();
 updateValue<{{declarations.print_variable_type(variable)}}>(__d, nest::{{names_namespace}}::_{{variable.get_symbol_name()}}, tmp_{{names.convert_to_cpp_name(variable.get_symbol_name())}});
+{%- endif %}
+
+{%- if variable.has_vector_parameter() %}
+{%- set vector_param = variable.get_vector_parameter() %}
+{%- if vector_param|int == 0 %}
```

Comment:
I'm a little confused as to what is happening here, why the special case for 0, and what happens when a parameter is a variable and cannot be cast to int? Also the if-statement on line 36 below I don't understand. Could you perhaps explain it for slow me or add some comments?